app/models/spectator_sport/session_window_analysis.rb

```diff
@@ -0,0 +1,17 @@
+module SpectatorSport
+  class SessionWindowAnalysis
+    def initialize(session_window)
+      @session_window = session_window
+    end
+
+    def visited_pages
+      @session_window.events.map { _1.page }.compact
+    end
+
+    def tags # simple naive implementation of tagging sessions
+      tags = []
+      tags << "server_error" if visited_pages.any? { _1.include?("500.html") }
```

Comment:
> In future, error page could probably emit custom event for simpler detection. I looked into custom events a bit, but so far wasn't able to use them.

Reply:
> The 500.html error page is rarely served directly.
Yes, you are right that this doesn't really make sense :sweat_smile:
 
> will want to create a custom controller/acton that raises an exception
I'm not sure what you mean by that. You mean for the demo app app?

Reply:
> You mean for the demo app app?

yes, for the demo app 😅

Reply:
I originally did that, but it does show dev error page instead of 500.html (as it's run in dev enviroment by default)
I found out we could force that using `config.consider_all_requests_local = false`, maybe for demo its sort of ok default behavior?

The question remains then - how to detect that the page shown is error.
Other than some middleware i can only think of rrweb custom event as a general solution.
Maybe you have some other ideas?

I can remove the error labeling from this PR until we solve that, unless we find some straightforward solution.

Reply:
Agree, will remove error tagging for now, and will let you know when PR will be ready for another look :)

Thanks a lot!

Reply:
Got it figured out: https://github.com/bensheldon/spectator_sport/pull/17

Reply:
Sorry, I'm super busy at work at the moment, so I didn't get to this.